addons/website_slides/models/slide_slide.py

```diff
@@ -489,6 +489,15 @@ def write(self, values):
             values['is_preview'] = True
             values['is_published'] = True
 
+        # if the slide type is changed, remove incompatible url or html_content
+        # done here to satisfy the SQL constraint
+        # using a stored-computed field or modifying the 'values' dict in place does not work
```

Comment:
it fail because of the write sequence

you can do something like this

```py
if 'slide_type' in values:
    if values['slide_type'] == 'webpage':
        values = {'url': False, **values}
    elif values['slide_type'] != 'webpage':
        values = {'html_content': False, **values}